### DISCUSSION TITLE
mpremote to run a file resident in a picow

### DISCUSSION THREAD

**Discussion by hcgrant** [micropython/micropython#12005] [RP2040 / Pico] [answered]
Is it possible to use mpremote to run code in the pico's ram?  
I can see that I can use  "mpremote run <filename> "  where the <filename>  exits in the local machine (in my case running Kubuntu). 
I've tried "mpremote run :<filename>" but that didnt work.
So .. 
any ideas?

**Comment by hcgrant** [comment]
Is it possible to use mpremote to run code in the pico's ram?
I can see that I can use "mpremote run filename" where the exits in the local machine (in my case running Kubuntu).
I've tried "mpremote run :filename" but that didnt work.
So ..
any ideas?

**Comment by bixb922** [comment]
If you have a file x.py in your microcontroller, you run that in the microcontroller with ```mpremote run x.py```
No : needed.

**Comment by robert-hh** [comment, marked as answer]
> Is it possible to use mpremote to run code in the pico's ram?

mpremote exec "import filename"

**Comment by hcgrant** [comment]
Many thanks robert-hh and bixb922 ... Both methods work thanks .. previously I tried    mpremote run filename    and that gave me an error something  like couldnt find filename hence I tried using ":" before the filename.

**Comment by bixb922** [comment]
You will also interested in ```mpremote mount . run main.py```
This will mount the current directory of your PC on the microcontroller and the run the main.py that's on the PC without having to transfer it to the microcontroller. Does not need all the imports copied. Very convenient! See https://docs.micropython.org/en/latest/reference/index.html

**Comment by bixb922** [comment]
You will also be interested in ```mpremote mount . run main.py```
This will mount the current directory of your PC on the microcontroller and run the main.py that's on the PC without having to transfer it to the microcontroller. Will also recognize imports without copying. Very convenient! See https://docs.micropython.org/en/latest/reference/index.html
